optee-client: Add CMake file for building libckteec

Origin: upstream, https://github.com/qualcomm/minkipc/commit/b938bceb35b21f195164c2233a98ccaed8fb4819
Backport of upstream optee-client/CMakeLists.txt from minkipc and adapt it to this repository layout.

diff -ruN optee_client-4.0.0.orig/CMakeLists.txt optee_client-4.0.0.new/CMakeLists.txt
--- optee_client-4.0.0.orig/CMakeLists.txt	2026-09-03 02:11:16.916257254 +0530
+++ optee_client-4.0.0.new/CMakeLists.txt	2026-09-03 02:13:05.142752766 +0530
@@ -1,11 +1,9 @@
-cmake_minimum_required(VERSION 3.4)
-project(optee_client C)
-
-# https://cmake.org/Wiki/CMake_Useful_Variables
-set(CMAKE_TOOLCHAIN_FILE CMakeToolchain.txt)
+cmake_minimum_required(VERSION 3.10)
+project(ckteec
+	VERSION 0.1.0
+	LANGUAGES C)
 
 option(CFG_WERROR "Build with -Werror" TRUE)
-option(WITH_TEEACL "Build libteeacl" TRUE)
 
 include(GNUInstallDirs)
 
@@ -38,12 +36,75 @@ if(CCACHE_FOUND)
     set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ccache)
 endif(CCACHE_FOUND)
 
-add_subdirectory(libteec)
-add_subdirectory(tee-supplicant)
-add_subdirectory(libckteec)
-if(WITH_TEEACL)
-    find_package(PkgConfig REQUIRED)
-    pkg_check_modules(uuid REQUIRED IMPORTED_TARGET uuid)
-    add_subdirectory(libteeacl)
-endif(WITH_TEEACL)
-add_subdirectory(libseteec)
+################################################################################
+# Packages
+################################################################################
+find_package(Threads REQUIRED)
+if(NOT THREADS_FOUND)
+	message(FATAL_ERROR "Threads not found")
+endif()
+
+include(GNUInstallDirs)
+
+################################################################################
+# Source files
+################################################################################
+set(SRC
+	libckteec/src/pkcs11_api.c
+	libckteec/src/ck_debug.c
+	libckteec/src/ck_helpers.c
+	libckteec/src/invoke_ta.c
+	libckteec/src/pkcs11_processing.c
+	libckteec/src/pkcs11_token.c
+	libckteec/src/serializer.c
+	libckteec/src/serialize_ck.c
+	src/teec_trace.c
+)
+
+################################################################################
+# Built library
+################################################################################
+add_library(ckteec SHARED ${SRC})
+
+set_target_properties(ckteec PROPERTIES
+	VERSION ${PROJECT_VERSION}
+	SOVERSION ${PROJECT_VERSION_MAJOR}
+)
+
+################################################################################
+# Flags always set
+################################################################################
+target_compile_definitions(ckteec
+	PRIVATE -D_GNU_SOURCE
+	PRIVATE -DBINARY_PREFIX="LT"
+)
+
+################################################################################
+# Optional flags
+################################################################################
+
+################################################################################
+# Public and private header and library dependencies
+################################################################################
+target_include_directories(ckteec
+	PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/libckteec/include>
+	$<INSTALL_INTERFACE:libckteec/include>
+	PRIVATE libckteec/src
+	PRIVATE include
+)
+
+target_link_libraries(ckteec
+	PRIVATE pthread
+	PRIVATE minkteec
+	PRIVATE minkadaptor
+	PRIVATE m
+)
+
+################################################################################
+# Install targets
+################################################################################
+install(TARGETS ckteec
+	DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
+
+add_subdirectory(libckteec/include)
